libs/k8s-client/src/BaseApi/PodWithLogStream.php

```diff
@@ -0,0 +1,42 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Keboola\K8sClient\BaseApi;
+
+use GuzzleHttp\Handler\CurlHandler;
+use GuzzleHttp\HandlerStack;
+use Kubernetes\API\Pod;
+use KubernetesRuntime\Client;
+use Psr\Http\Message\ResponseInterface;
+
+class PodWithLogStream extends Pod
+{
+    public function __construct($namespace = 'default')
+    {
+        $this->client = Client::getInstance();
+        $this->namespace = $namespace;
+    }
+
+    public function readLog(string $namespace, string $name, array $queries = [])
+    {
+        $handler = new CurlHandler();
+        $handlerStack = HandlerStack::create($handler);
+
+        /** @var ResponseInterface $response */
+        $response = $this->client->request(
+            'get',
+            sprintf('/api/v1/namespaces/%s/pods/%s/log', $namespace, $name),
+            [
+                'query' => $queries,
+                'handler' => $handlerStack,
+            ],
+        );
+
+        if ($response->getStatusCode() >= 200 && $response->getStatusCode() < 300) {
```

Comment:
Tady bych mozna byl jeste defenzivnejsi. Podle docs ma ten endpoint vrace `200` (https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.31/#read-log-pod-v1-core), takze bych klidne hlidal presne `200`.

Reply:
To jsem tam měl ale padal pak test testReadLogWhenNodeSelectorIsUnsatisfied()

Reply:
Koukam, protoze to muze vratit `204`.. ok, tak to asi nechme